spec/features/session_life_spec.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Session cookies', type: :feature do
+  let(:mock_user) { 'patron1' }
+
+  before do
+    login_permanently_as username: 'PATRON1', patron_key: mock_user
+    visit root_path
+  end
+
+  after do
+    Warden::Manager._on_request.clear
+  end
+
+  describe 'the session cookies', js: true do
+    context 'when starting a fresh session' do
+      it 'sets session cookies' do
+        session_began_cookie = page.driver.browser.manage.cookie_named(:session_began)
+        last_active_cookie = page.driver.browser.manage.cookie_named(:last_active)
+
+        expect(session_began_cookie[:value]).to be_present
+        expect(last_active_cookie[:value]).to be_present
+      end
+    end
+
+    xcontext 'when a user has not done anything in the application for a duration greater than the last_active value' do
```

Comment:
I wasn't able to get a test that properly tests these scenarios, but at least they are documented. The warden test helper for logging in always wants to start a new session with every request.